comp/dogstatsd/server/parse.go

```diff
@@ -270,66 +270,35 @@ func (p *parser) parseFloat64List(rawFloats []byte) ([]float64, error) {
 	return values, nil
 }
 
-// resolveContainerIDFromLocalData returns the container ID for the given Local Data.
-// The Local Data is a list that can contain one or two (split by a ',') of either:
-// * "ci-<container-id>" for the container ID.
-// * "in-<cgroupv2-inode>" for the cgroupv2 inode.
-// Possible values:
-// * "<container-id>"
-// * "ci-<container-id>"
-// * "ci-<container-id>,in-<cgroupv2-inode>"
-func (p *parser) resolveContainerIDFromLocalData(rawLocalData []byte) []byte {
-	// Remove prefix from Local Data
-	localData := rawLocalData[len(localDataPrefix):]
-
-	var containerID []byte
-	var containerIDFromInode []byte
-
-	if bytes.Contains(localData, []byte(",")) {
-		// The Local Data can contain a list
-		items := bytes.Split(localData, []byte{','})
-		for _, item := range items {
-			if bytes.HasPrefix(item, containerIDPrefix) {
-				containerID = item[len(containerIDPrefix):]
-			} else if bytes.HasPrefix(item, inodePrefix) {
-				containerIDFromInode = p.resolveContainerIDFromInode(item[len(inodePrefix):])
-			}
-		}
-		if containerID == nil {
-			containerID = containerIDFromInode
-		}
-	} else {
-		// The Local Data can contain a single value
-		if bytes.HasPrefix(localData, containerIDPrefix) { // Container ID with new format: ci-<container-id>
-			containerID = localData[len(containerIDPrefix):]
-		} else if bytes.HasPrefix(localData, inodePrefix) { // Cgroupv2 inode format: in-<cgroupv2-inode>
-			containerID = p.resolveContainerIDFromInode(localData[len(inodePrefix):])
-		} else { // Container ID with old format: <container-id>
-			containerID = localData
-		}
+// resolveContainerIDFromInode returns the container ID for the given cgroupv2 inode.
+// TODO: This method will be removed when Inode resolution has been moved to the Tagger.
+func (p *parser) resolveLocalData(rawLocalData []byte) (origindetection.LocalData, error) {
+	localDataString := string(rawLocalData)
+
+	var localData origindetection.LocalData
+	var err error
+	localData, err = origindetection.ParseLocalData(localDataString)
+	if err != nil {
+		log.Errorf("failed to parse c: field containing Local Data %s: %v", localDataString, err)
```

Comment:
Same as `rawExternalData`:

```suggestion
		log.Errorf("failed to parse c: field containing Local Data %q: %v", localDataString, err)
```